### ISSUE TITLE
[MPS] layer_norm_backward silently returns incorrect, nondeterministic parameter gradients

### ISSUE THREAD

**Issue by jhavukainen** [pytorch/pytorch#183825]
### 🐛 Describe the bug

layer_norm_backward on the MPS backend silently returns incorrect, nondeterministic parameter gradients (weight.grad, bias.grad) when the input tensor does not require grad but the layer-norm parameters do. CPU returns the correct gradients in both requires_grad=True and requires_grad=False configurations. MPS only returns correct gradients when input.requires_grad=True.

Reproducer:
```python
import platform
import torch
import torch.nn.functional as F


def run(x_requires_grad: bool):
    torch.manual_seed(0)
    x = torch.randn(2, 4, requires_grad=x_requires_grad)
    weight = torch.ones(4, requires_grad=True)
    bias = torch.zeros(4, requires_grad=True)

    def compute(device):
        xd = x.to(device)
        wd = weight.detach().clone().to(device).requires_grad_(True)
        bd = bias.detach().clone().to(device).requires_grad_(True)
        out = F.layer_norm(xd, (4,), wd, bd, 1e-5)
        out.sum().backward()
        return wd.grad.detach().cpu().tolist(), bd.grad.detach().cpu().tolist()

    cpu_w, cpu_b = compute("cpu")
    mps_w, mps_b = compute("mps") if torch.backends.mps.is_available() else (None, None)
    return cpu_w, cpu_b, mps_w, mps_b


def fmt(v):
    return "[" + ", ".join(f"{x:+.4f}" for x in v) + "]"


def _close(a, b, tol=1e-4):
    return all(abs(x - y) <= tol for x, y in zip(a, b))


def main():
    print(f"PyTorch:  {torch.__version__}")
    print(f"Platform: {platform.platform()}\n")

    for req_grad in (True, False):
        cpu_w, cpu_b, mps_w, mps_b = run(req_grad)
        print(f"=== x.requires_grad={req_grad} ===")
        print(f"  cpu  w.grad = {fmt(cpu_w)}")
        print(f"  mps  w.grad = {fmt(mps_w)}")
        print(f"  cpu  b.grad = {fmt(cpu_b)}")
        print(f"  mps  b.grad = {fmt(mps_b)}")
        ok = _close(cpu_w, mps_w) and _close(cpu_b, mps_b)
        print(f"  -> {'MATCH' if ok else 'MISMATCH — bug'}\n")


if __name__ == "__main__":
    main()
```

Output when running on the current nightly:

```
PyTorch:  2.13.0.dev20260514
Platform: macOS-26.4-arm64-arm-64bit

=== x.requires_grad=True ===
  cpu  w.grad = [+0.3771, -1.2932, -0.7740, +1.6900]
  mps  w.grad = [+0.3771, -1.2932, -0.7740, +1.6900]
  cpu  b.grad = [+2.0000, +2.0000, +2.0000, +2.0000]
  mps  b.grad = [+2.0000, +2.0000, +2.0000, +2.0000]
  -> MATCH

=== x.requires_grad=False ===
  cpu  w.grad = [+0.3771, -1.2932, -0.7740, +1.6900]
  mps  w.grad = [-0.0907, -0.3104, +0.0000, +0.0000]
  cpu  b.grad = [+2.0000, +2.0000, +2.0000, +2.0000]
  mps  b.grad = [+0.0000, +0.0000, +0.0000, +0.0000]
  -> MISMATCH — bug
```

### Versions

PyTorch:  2.13.0.dev20260514
Platform: macOS-26.4-arm64-arm-64bit

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93 @kulinseth @malfet @DenisVieriu97 @aditvenk
